worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/TaskStowingBulkWorker.java

```diff
@@ -0,0 +1,231 @@
+                    validatedWorkerTasks.size(), workerTasks.size(), validatedWorkerTasks.size());
+        try {
+            databaseClient.insertStowedTasks(
+                partitionIdList,
+                jobIdList,
+                taskClassifierList,
+                taskApiVersionList,
+                taskDataList,
+                taskStatusList,
+                contextList,
+                toList,
+                trackingInfoList,
+                sourceInfoList,
+                correlationIdList);
+
+            validatedWorkerTasks.forEach(workerTask -> {
+                workerTask.setResponse(createSuccessResultNoOutputToQueue());
+            });
+        } catch (final Exception exception) {
+            LOGGER.error(TaskStowingWorkerFailure.FAILED_TO_WRITE_TO_DATABASE, exception);
+            validatedWorkerTasks.forEach(workerTask -> {
+                workerTask.setResponse(createFailureResult(TaskStowingWorkerFailure.FAILED_TO_WRITE_TO_DATABASE, exception));
+            });
+        }
+    }
+
+    private WorkerResponse createSuccessResultNoOutputToQueue()
+    {
+        return WorkerResponseFactory.createSuccessResultNoOutputToQueue(WORKER_IDENTIFIER, WORKER_API_VERSION);
+    }
+
+    private WorkerResponse createFailureResult(final String failure)
+    {
+        return WorkerResponseFactory.createFailureResult(outputQueue, WORKER_IDENTIFIER, WORKER_API_VERSION, codec, failure);
+    }
+
+    private WorkerResponse createFailureResult(final String failure, final Throwable cause)
+    {
+        return WorkerResponseFactory.createFailureResult(outputQueue, WORKER_IDENTIFIER, WORKER_API_VERSION, codec, failure, cause);
+    }
+
+    private static boolean anyListsEmptyOrNotSameSize(
+        final List<WorkerTask> validatedWorkerTasks,
+        final List<String> partitionIdList,
+        final List<String> jobIdList,
+        final List<String> taskClassifierList,
+        final List<Integer> taskApiVersionList,
+        final List<byte[]> taskDataList,
+        final List<String> taskStatusList,
+        final List<byte[]> contextList,
+        final List<String> toList,
+        final List<byte[]> trackingInfoList,
+        final List<byte[]> sourceInfoList,
+        final List<String> correlationIdList)
+    {
+        final List<List<? extends Object>> lists = new ArrayList<>();
+        lists.add(validatedWorkerTasks);
+        lists.add(partitionIdList);
+        lists.add(jobIdList);
+        lists.add(taskClassifierList);
+        lists.add(taskApiVersionList);
+        lists.add(taskDataList);
+        lists.add(taskStatusList);
+        lists.add(contextList);
+        lists.add(toList);
+        lists.add(trackingInfoList);
+        lists.add(sourceInfoList);
+        lists.add(correlationIdList);
+
+        return !lists.stream().allMatch(list -> !list.isEmpty() && (list.size() == lists.get(0).size()));
```

Comment:
This is very minor but by getting the size first you can check for empty and avoid the repeated lookups:

    final int firstListSize = validatedWorkerTasks.size();
    return firstListSize == 0
        ? true
        : lists.stream().anyMatch(list -> list.size() != firstListSize);

Also you should generally use `anyMatch()` rather than `!allMatch()` (or use `allMatch()` rather than `!anyMatch()`) as these are easier to read without the not - that is why they give you both.